Meter.c

```diff
@@ -60,12 +60,12 @@ static void TextMeterMode_draw(Meter* this, int x, int y, int w) {
    }
    attrset(CRT_colors[RESET_COLOR]);
 
-   int captionLen = strlen(caption);
-   w -= captionLen;
-   if (w < 1) {
+   int captionWidth = (int)strnlen(caption, (unsigned int)w);
```

Comment:
Add an assert for `w >= 0` and cast the second argument to `size_t`.

Reply:
1. Nice catch on missing the `w >= 0` check. I add it now as a conditional (rather than an assertion).
2. The `(unsigned int)` cast was meant to be an optimization hint. I intended the cast to be `(size_t)(unsigned int)w`, but I except a decent compiler should be able to figure that out. I now make the cast implicit instead. (If a compiler can throw a warning on that, it's a compiler bug.)

Reply:
And [here is a GCC bug report about sign extension missed optimization](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=120434).